src/bar/FilterSelect.vue

```diff
@@ -94,12 +100,23 @@
                 if (!this.isXAxisOrDisagg) {
                     //When we go from multi-select to single-select, update 'selected'
                     if (this.selected.length > 1) {
-                        this.selected = [this.selected[0]];
+                        this.selected = this.defaultValue.defaultSelections
+                            ? this.defaultValue.defaults
+                            : [this.selected[0]];
                     }
                     if (this.selected.length == 0) {
-                        this.selected.push(this.options[0]);
+                        this.defaultValue.defaultSelections
+                            ? this.selected = this.defaultValue.defaults
+                            : this.selected.push(this.options[0]);
                     }
                     this.$emit("input", this.selected);
+                } else {
+                    //When we go from single-select or multi-select
+                    //to multi-select, update 'selected' with default selection
+                    if (this.defaultValue.defaultSelections) {
+                        this.selected = this.defaultValue.defaults;
+                        this.$emit("input", this.selected);
+                    }
```

Comment:
I have considered doing that but the PR is not meant to update selected when value changes but update `this.selected` when disaggregate or isXasis changes. 

The reason for attempting to do this is because when indicator changes, we want to apply default selections so when element is moving from multi-select to single-select, the current behaviour selects the first index of the selected values. However, since we now have default values we no longer want to select first index of the selected values. The same goes to moving from single to multiple values.

Reply:
> the PR is not meant to update selected when value changes but update this.selected when disaggregate or isXasis changes.

That was working before, though, wasn't it? The filter was always liable to change its isAxisOrDisaggregate value, due to user selecting new x-axis or disaggregate by filter. The difference now is that value can change externally too. 

> However, since we now have default values we no longer want to select first index of the selected values. The same goes to moving from single to multiple values.

But aren't the `defaultValue`s actually just `value` as far as this control is concerned? In the parent control, `BarchartWithFilters`, the defaults in  defaultValue are being set to `selections.selectedFilterOptions[filter.id]` - but that's exactly the same as what's going into `value` via `getSelectedFilterOptions`:
https://github.com/reside-ic/vue-charts/blob/master/src/bar/BarChartWithFilters.vue#L33
https://github.com/reside-ic/vue-charts/blob/master/src/bar/BarChartWithFilters.vue#L266

Both of these will be updated when new selections are provided from HINT. So it's really not clear to me why `defaultValue` is required.

Reply:
Looks like you were right all along. I have added suggested changes.